src/sw/redis++/connection.cpp

```diff
@@ -93,6 +99,54 @@ auto ConnectionOptions::_split_path(const std::string &path) const
     return std::make_tuple(path, 0);
 }
 
+template <>
+void ConnectionOptions::_set_optional_key<bool>(const std::string &options_uri, const std::string &key, bool &field) const {
+    if (auto option = _extract_string_option(options_uri, key)) {
+        if (option != "true" && option != "false") {
+            throw Error("Bad formatted URI: unable to parse option '" + *option + "' for " + key);
+        }
+        field = option == "true" ? true : false;
+    }
+    // that option was not given.
+}
+
+template <>
+void ConnectionOptions::_set_optional_key<std::chrono::milliseconds>(const std::string &options_uri, const std::string &key, std::chrono::milliseconds &field) const {
+    if (auto option = _extract_string_option(options_uri, key)) {
+        try {
+            field = std::chrono::milliseconds(std::stoi(*option));
+        } catch (const std::exception &) {
+            throw Error("Bad formatted URI: unable to parse option '" + *option + "' for " + key);
+        }
+    }
+    // that option was not given.
+}
+
+std::optional<std::string> ConnectionOptions::_extract_string_option(const std::string &options_uri, const std::string &key) const {
+    if (auto pos = options_uri.find(key); pos != std::string::npos) {
+        // take string after key and equal sign. Equal sign must follow the key, if not we will have a bad formatting exception
+        auto option = options_uri.substr(pos + key.length() + 1); 
+        auto option_end = option.find("&");
+        return option.substr(0, option_end != std::string::npos ? option_end : option.length()); // the last option has no '&'
```

Comment:
we cannot return a std::string, since the return value is `Optional<std::string>`.

Reply:
I thought that C++17 was a requirement for this repo, sorry about that